src/pipeline/src/etl/processor/digest.rs

```diff
@@ -0,0 +1,433 @@
+            _ => PresetPatternInvalidSnafu { pattern }.fail(),
+        }
+    }
+}
+
+impl PresetPattern {
+    fn regex(&self) -> Regex {
+        match self {
+            PresetPattern::Numbers => Regex::new(r"\d+").unwrap(),
+            PresetPattern::Quoted => Regex::new(r#"[\"'“”‘’][^\"'“”‘’]*[\"'“”‘’]"#).unwrap(),
+            PresetPattern::Bracketed => Regex::new(r#"[({\[<「『【〔［｛〈《][^(){}\[\]<>「」『』【】〔〕［］｛｝〈〉《》]*[)}\]>」』】〕］｝〉》]"#).unwrap(),
+            PresetPattern::Uuid => Regex::new(r"\b[0-9a-fA-F]{8}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{4}\b-[0-9a-fA-F]{12}\b").unwrap(),
+            PresetPattern::Ip => Regex::new(r"((\d{1,3}\.){3}\d{1,3}(:\d+)?|(\[[0-9a-fA-F:]+\])(:\d+)?)").unwrap(),
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+pub struct DigestProcessorBuilder {
+    fields: Fields,
+    patterns: Vec<Regex>,
+    ignore_missing: bool,
+}
+
+impl ProcessorBuilder for DigestProcessorBuilder {
+    fn output_keys(&self) -> HashSet<&str> {
+        self.fields
+            .iter()
+            .map(|f| f.target_or_input_field())
+            .collect()
+    }
+
+    fn input_keys(&self) -> HashSet<&str> {
+        self.fields.iter().map(|f| f.input_field()).collect()
+    }
+
+    fn build(self, intermediate_keys: &[String]) -> Result<ProcessorKind> {
+        self.build(intermediate_keys).map(ProcessorKind::Digest)
+    }
+}
+
+impl DigestProcessorBuilder {
+    fn build(self, intermediate_keys: &[String]) -> Result<DigestProcessor> {
+        let mut real_fields = vec![];
+        for field in self.fields.into_iter() {
+            let input = OneInputOneOutputField::build(
+                "digest",
+                intermediate_keys,
+                field.input_field(),
+                field.target_or_input_field(),
+            )?;
+            real_fields.push(input);
+        }
+        Ok(DigestProcessor {
+            fields: real_fields,
+            ignore_missing: self.ignore_missing,
+            patterns: self.patterns,
+        })
+    }
+}
+
+/// Computes a digest (hash) of the input string.
+#[derive(Debug, Default)]
+pub struct DigestProcessor {
+    fields: Vec<OneInputOneOutputField>,
+    ignore_missing: bool,
+    patterns: Vec<Regex>,
+}
+
+impl DigestProcessor {
+    fn remove_quoted_content(&self, val: &str) -> String {
+        let re = Regex::new(r#""[^"]*""#).unwrap();
+        re.replace_all(val, "").to_string()
+    }
+
+    fn process_string(&self, val: &str) -> Result<Value> {
+        let mut input = val;
+        let mut result = None;
+        for pattern in self.patterns.iter() {
+            let new_result = pattern.replace_all(input, "");
+            if new_result != input {
+                result = Some(new_result.to_string());
+                input = result.as_ref().unwrap();
+            }
+        }
+
+        if let Some(result) = result {
+            Ok(Value::String(result))
+        } else {
+            Ok(Value::String(val.to_string()))
+        }
```

Comment:
Use Cow to avoid unnecessary memory allocation.

```suggestion
        use std::borrow::Cow;

        let mut input = Cow::from(val);
        for pattern in &self.patterns {
            if let Cow::Owned(new_string) = pattern.replace_all(&input, "") {
                input = Cow::Owned(new_string);
            }
        }

        Ok(Value::String(input.into_owned()))
```